gammapy/scripts/image_pipe.py

```diff
@@ -175,13 +222,135 @@ def make_maps(self, radius, bkg_norm=True):
+        ncounts_min : int
+            Minimum counts required for one observation
         """
-        self.make_total_counts()
-        self.make_total_bkg(bkg_norm)
+        self.make_total_counts(ncounts_min)
+        self.make_total_bkg(bkg_norm, ncounts_min)
         self.make_significance(radius)
+        self.make_total_excess()
+        self.make_total_exposure(spectral_index, exposure_weighted_spectrum, ncounts_min)
 
     def make_psf(self):
         log.info('Making PSF ...')
 
-    def make_exposure(self):
-        log.info('Making EXPOSURE ...')
+    def make_total_excess(self):
+        """Compute excess between counts and bkg map."""
+        self.maps["total_excess"] = SkyMap.empty_like(self.empty_image)
+        excess = self.maps["total_counts"].data - self.maps["total_bkg"].data
+        self.maps["total_excess"].data = excess
+
+    def make_exposure(self, obs_id, spectral_index=2.3):
+        """Compute the exposure map for one observation. Excess/exposure will give the differential flux
+        at the energy eref
+
+        Parameters
+        ----------
+        obs_id : int
+            Number of the observation
+        spectral_index : int
+
+        """
+        # TODO: should be re-implemented using the exposure_cube function
+        exposure = SkyMap.empty_like(self.empty_image)
+
+        # Determine offset value for each pixel of the map
+        xpix_coord_grid, ypix_coord_grid = exposure.coordinates(coord_type='pix')
+        # calculate pixel offset from center (in world coordinates)
+        coord = pixel_to_skycoord(xpix_coord_grid, ypix_coord_grid, exposure.wcs, origin=0)
+        center = SkyCoord.from_pixel(self.header["NAXIS1"] / 2., self.header["NAXIS2"] / 2., exposure.wcs)
+        offset = coord.separation(center)
+
+        obs = self.data_store.obs(obs_id=obs_id)
+        livetime = obs.observation_live_time_duration
+
+        # 2D Exposure computation on the self.energy_range and on an offset_tab
+        energy = EnergyBounds(np.linspace(self.energy_band[0].value, self.energy_band[1].value, 100),
+                              self.energy_band.unit)
+        energy_band = energy.bands
+        energy_bin = energy.lin_centers
+        eref = EnergyBounds(self.energy_band).log_centers
+        spectrum = (energy_bin / eref) ** (-spectral_index)
+        aeff2d = obs.aeff
+        offset_tab = Angle(np.linspace(self.offset_band[0].value, self.offset_band[1].value, 10), self.offset_band.unit)
+        exposure_tab = np.sum(aeff2d.evaluate(offset_tab, energy_bin).to("cm2") * spectrum * energy_band, axis=1)
+
+        # Interpolate for the offset of each pixel
+        f = interpolate.interp1d(offset_tab, exposure_tab, bounds_error=False, fill_value=0)
+        exposure.data = f(offset)
+        exposure.data *= livetime
+        exposure.data[offset >= self.offset_band[1]] = 0
+        self.maps["exposure"] = exposure
+
+    def make_exposure_weighted_spectrum(self, obs_id, spectral_index=2.3):
```

Comment:
So the difference between `make_exposure` and `make_exposure_weighted_spectrum` is just that the first returns diff flux at given energy, and the other integral flux in energy band, and the difference is one constant factor for the whole image?

If yes, the second version `make_exposure_weighted_spectrum` and all the duplicated code there should be removed.
The caller can simply compute and apply that factor herself.
Or you provide a convenience method on this class to do it, but calling the other one or assuming it's called after, not by duplicating the code there.